### PR TITLE
Minor fixes: to list_files and mkdir in fpm_filesystem

### PR DESCRIPTION
**Pull request by LKedward**
As identified by @urbanjost, there are inconsistencies in the behaviour of our `mkdir` and `list_files` filesystem routines on Windows and Linux.

PR implements the following fixes:

- `mkdir` will not fail if dir already exists;
- `list_files` will include hidden files but exclude '.' and '..'

### CODE REVIEW

**Review by milancurcic — APPROVED** [review]

**Comment by milancurcic** [comment]
Bump. Let's get one more review here before merging.

**Comment by interkosmos** [comment]
Looks fine.

**Comment by milancurcic** [comment]
Thank you! Will merge.

**Pull request merged**
